### DISCUSSION TITLE
Correct working directory for sidecar binaries

### DISCUSSION THREAD

**Discussion by boan-anbo** [tauri-apps/tauri#5206] [Q&A]
Hi there! I have two binaries embedded, one is a single-file executable, and another is a dotnet program with multiple files.

The single executable works well.

The second with multiple files, however, cannot be started with `Command::new_sidecar("db").spawn().` 

Then I found out the command's working directory (`current_dir()`) needs to be set to `src-tauri\\target\\debug`. 

Since I cannot rely on using an absolute path like `src-tauri\\target\\debug` on release, **how can I set the `current_dir()` for the sidecar binary correctly?**

I tried `std::env::current_dir` but that, when debugging, points to ``src-tauri,`` not were the binary is actually located, i.e. src-tauri\\target\\debug.`

Thanks!

**Comment by FabianLars** [comment]
> Since I cannot rely on using an absolute path like src-tauri\\target\\debug on release, how can I set the current_dir() for the sidecar binary correctly?

That depends on what the working directory needs to be in release for your app..

I know of a few people (including myself...) who use sidecars with runtime dependencies by moving the exe and its dependencies directly next to the tauri.conf.json file in the src-tauri dir, and then adding the exe to `externalBin` and the dependencies to `resources`. This way the files are in both, debug and release next to each other with the cwd being set currectly.

**Reply by boan-anbo** [reply]
> > Since I cannot rely on using an absolute path like src-tauri\target\debug on release, how can I set the current_dir() for the sidecar binary correctly?
> 
> That depends on what the working directory needs to be in release for your app..
> 
> I know of a few people (including myself...) who use sidecars with runtime dependencies by moving the exe and its dependencies directly next to the tauri.conf.json file in the src-tauri dir, and then adding the exe to `externalBin` and the dependencies to `resources`. This way the files are in both, debug and release next to each other with the cwd being set currectly.

Great tips! Thanks so much!
